### PR TITLE
Vslt/vslt 367 customer doc

### PR DESCRIPTION
**Pull request by xwa130saucer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,12 +42,12 @@ module.exports = {
...
 
 Run the following command:
 
-```
-$ npm run test-screener
+```bash
+npm run test-screener
 ```
 
 ### Testing Interactions
```
Comment (loyalchow) [inline README.md]: I see you've changed headings. This should also be `##` then

File: README.md [hunk 2]
```diff
@@ -212,10 +219,6 @@ module.exports = {
 | microsoftedge | 17.17134 | requires [Sauce Labs](https://screener.io/v2/docs/sauce) Integration |
 | safari | 11.1 | requires [Sauce Labs](https://screener.io/v2/docs/sauce) Integration |
 
-
-
-
-
 ### Additional Configuration Options
```
Comment (loyalchow) [inline README.md]: This should now also be `##`

File: README.md [hunk 3]
```diff
@@ -287,47 +301,113 @@ module.exports = {
...
-    - Example:
+  - Example:
+
     ```javascript
     vsts: {
       instance: 'myproject.visualstudio.com'
     }
     ```
+
+### Sauce Connect Integration
```
Comment (loyalchow) [inline README.md]: Please move this into Cross Browser Testing as a sub-section

File: README.md [hunk 4]
```diff
@@ -287,47 +301,113 @@ module.exports = {
...
+
     ```javascript
     sauce: {
       username: 'sauce_user',
       accessKey: 'sauce_access_key',
       maxConcurrent: 10, // optional available concurrency you have from Sauce Labs
       extendedDebugging: true, // optional
-      tunnelIdentifier: 'MyTunnel01' // optional
+      tunnelIdentifier: 'MyTunnel01', // optional
+      launchSauceConnect: true // optional
```
Comment (loyalchow) [inline README.md]: Please add a note here to view "Sauce Connect" for more information

File: README.md [hunk 5]
```diff
@@ -287,47 +301,113 @@ module.exports = {
...
+
+You could run your tests through sauce connect tunnel by setting the flag `launchSauceConnect: true`:
+
+  ```javascript
+  sauce: {
+    username: 'sauce_user',
+    accessKey: 'sauce_access_key',
+    maxConcurrent: 10, // optional available concurrency you have from Sauce Labs
+    extendedDebugging: true, // optional
+    launchSauceConnect: true // optional
```
Comment (loyalchow) [inline README.md]: Remove the `extendedDebugging` option to simplify this example

File: README.md [hunk 6]
```diff
@@ -287,47 +301,113 @@ module.exports = {
...
+
     ```javascript
     vsts: {
       instance: 'myproject.visualstudio.com'
     }
     ```
+
+### Sauce Connect Integration
+
+You could run your tests through sauce connect tunnel by setting the flag `launchSauceConnect: true`:
```
Comment (loyalchow) [inline README.md]: "When using Sauce Labs browsers, you have the option to use the Sauce Connect tunnel by..."

"When enabled, Sauce Connect will be launched and managed by this module, and assigned a unique tunnel identifier."

File: README.md [hunk 7]
```diff
@@ -287,47 +301,113 @@ module.exports = {
...
+  sauce: {
+    username: 'sauce_user',
+    accessKey: 'sauce_access_key',
+    maxConcurrent: 10, // optional available concurrency you have from Sauce Labs
+    extendedDebugging: true, // optional
+    launchSauceConnect: true // optional
+  }
+  ```
+
+The sauce connect tunnel will generate `sauce-connect.log` log file for debugging purpose at the root directory of your `screener-runner` project.
```
Comment (loyalchow) [inline README.md]: Let's simplify this and the below, and simply add a "Sauce Connect Notes" section that includes a bullet on each:

* Must use Sauce browsers
* Where SC logs are stored
* That SC tunnel cannot be used the `tunnel` option
* SC ports
* Link to SC troubleshooting docs in Sauce Labs

**Review by loyalchow — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (xwa130saucer) [reply README.md]: addressed

File: README.md (hunk 2 above)
Comment (xwa130saucer) [reply README.md]: addressed

File: README.md (hunk 3 above)
Comment (xwa130saucer) [reply README.md]: addressed

File: README.md (hunk 4 above)
Comment (xwa130saucer) [reply README.md]: addressed

File: README.md (hunk 5 above)
Comment (xwa130saucer) [reply README.md]: addressed

File: README.md (hunk 6 above)
Comment (xwa130saucer) [reply README.md]: addressed

File: README.md (hunk 7 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md [hunk 8]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
+  ```javascript
+  sauce: {
+    username: 'sauce_user',
+    accessKey: 'sauce_access_key',
+    maxConcurrent: 10, // optional available concurrency you have from Sauce Labs
+    launchSauceConnect: true // optional,
+  }
+  ```
 
+#### Sauce Connect Notes
```
Comment (loyalchow) [inline README.md]: Change to: `Important Notes on Sauce Connect`

File: README.md [hunk 9]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
 
+- Must use Sauce browsers
```
Comment (loyalchow) [inline README.md]: Please change this bullet to "Sauce Connect Integration requires all browsers to be Sauce Labs Browsers. An error is thrown when using non-Sauce browsers."

File: README.md [hunk 10]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
 
-### Additional Configuration Options
+  When you run your tests in sauce labs, you could only run your tests inside sauce browsers, which means that you have to specify both the name of the browser and the version of the browser up to one digit after the decimal dot:
```
Comment (loyalchow) [inline README.md]: Please remove this and config

File: README.md [hunk 11]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
+        {
+          browserName: 'internet explorer',
+          version: '11.0'
+        }
+      ]
+    ```
+
+  If you don't follow the rules, the runner will throw you the error: `Only Sauce Labs browsers with version can be used when launchSauceConnect flag is enabled`.
+
+- Where Sauce Connect logs are stored
```
Comment (loyalchow) [inline README.md]: Please change this to: "Logs for Sauce Connect are saved in the root of your project under `sauce-connect.log` for debugging purposes."

File: README.md [hunk 12]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
+
+  The sauce connect tunnel will generate `sauce-connect.log` log file for debugging purpose at the root directory of your project.
+
+- Sauce tunnel can not be used together with ngrok tunnel
```
Comment (loyalchow) [inline README.md]: Please change to: "Sauce Connect tunnel cannot be used together with the `tunnel` option."

File: README.md [hunk 13]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
+
+    ```javascript
+      tunnel: {
+        host: 'localhost:3000'
+      }
+    ```
+
+  when you enable `launchSauceConnect`. If you don't follow the rules, the runner will throw you the error: `tunnel option cannot be set when launchSauceConnect flag is enabled`.
+
+- `launchSauceConnect` and `tunnelIdentifier`
```
Comment (loyalchow) [inline README.md]: Please change to: "A unique `tunnelIdentifier` is automatically generated for you when using the Sauce Connect Integration. An error is thrown when `tunnelIdentifier` is set manually.

File: README.md [hunk 14]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
+
+  Please note that if you turn `launchSauceConnect` to be true, the runner will set a unique `tunnelIdentifier` for you insead of setting by yourself. If you don't follow the rules, the runner will throw you the error: `tunnelIdentifier cannot be set when launchSauceConnect flag is enabled`.
+
+- Sauce Connect valid ports
```
Comment (loyalchow) [inline README.md]: Please change to: "When running Sauce Connect tunnel on your localhost, please note that Sauce Connect only supports a limited set of valid ports."

File: README.md [hunk 15]
```diff
@@ -212,69 +219,142 @@ module.exports = {
...
+
+  When you run your tests in your localhost, please be cautious that sauce connect only supports these [valid ports](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS#SauceConnectProxyFAQs-CanIAccessApplicationsonlocalhost?).
+
+- Further question
```
Comment (loyalchow) [inline README.md]: Please change to: "For additional information on Sauce Connect please refer to the Sauce Connect FAQ and Sauce Connect Troubleshooting documentation."

**Review by loyalchow — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 9 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 10 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 11 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 12 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 13 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 14 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md (hunk 15 above)
Comment (xwa130saucer) [reply README.md]: changed

File: README.md [hunk 16]
```diff
@@ -212,69 +219,123 @@ module.exports = {
...
+    launchSauceConnect: true // optional,
+  }
+  ```
 
+#### Important Notes on Sauce Connect
 
+- Sauce Connect Integration requires all browsers to be Sauce Labs Browsers. An error is thrown when using non-Sauce browsers.
 
-### Additional Configuration Options
+  If you don't follow the rules, the runner will throw you the error: `Only Sauce Labs browsers with version can be used when launchSauceConnect flag is enabled`.
```
Comment (loyalchow) [inline README.md]: Please remove this - I don't think it's necessary, as we mention an error above already

File: README.md [hunk 17]
```diff
@@ -212,69 +219,123 @@ module.exports = {
...
+
+- Logs for Sauce Connect are saved in the root of your project under `sauce-connect.log` for debugging purposes.
+
+  The sauce connect tunnel will generate `sauce-connect.log` log file for debugging purpose at the root directory of your project.
```
Comment (loyalchow) [inline README.md]: Please remove this. Line above covers it

File: README.md [hunk 18]
```diff
@@ -212,69 +219,123 @@ module.exports = {
...
+
+- Sauce Connect tunnel cannot be used together with the `tunnel` option.
+
+  `ngrok` tunnel is forbidden when you run the sauce connect tunnel, which means you can not set
```
Comment (loyalchow) [inline README.md]: Please remove this and below. Just keep first line of bullet. Clients do not know what "ngrok" is

File: README.md [hunk 19]
```diff
@@ -212,69 +219,123 @@ module.exports = {
...
+      tunnel: {
+        host: 'localhost:3000'
+      }
+    ```
+
+  when you enable `launchSauceConnect`. If you don't follow the rules, the runner will throw you the error: `tunnel option cannot be set when launchSauceConnect flag is enabled`.
+
+- A unique `tunnelIdentifier` is automatically generated for you when using the Sauce Connect Integration. An error is thrown when `tunnelIdentifier` is set manually.
+
+  Please note that if you turn `launchSauceConnect` to be true, the runner will set a unique `tunnelIdentifier` for you insead of setting by yourself. If you don't follow the rules, the runner will throw you the error: `tunnelIdentifier cannot be set when launchSauceConnect flag is enabled`.
```
Comment (loyalchow) [inline README.md]: Please remove this. Bullet line already covers

File: README.md [hunk 20]
```diff
@@ -212,69 +219,123 @@ module.exports = {
...
+
+- When running Sauce Connect tunnel on your localhost, please note that Sauce Connect only supports a limited set of valid ports.
+
+  When you run your tests in your localhost, please be cautious that sauce connect only supports these [valid ports](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS#SauceConnectProxyFAQs-CanIAccessApplicationsonlocalhost?).
+
+- For additional information on Sauce Connect please refer to the Sauce Connect FAQ and Sauce Connect Troubleshooting documentation.
+
+  If you have further question, you could visit [Sauce Connect FAQ](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS) and [Sauce Connect troubleshooting](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+Troubleshooting) pages.
```
Comment (loyalchow) [inline README.md]: Please remove this, and add links to the first bullet line

File: README.md [hunk 21]
```diff
@@ -212,69 +219,123 @@ module.exports = {
...
+
+  when you enable `launchSauceConnect`. If you don't follow the rules, the runner will throw you the error: `tunnel option cannot be set when launchSauceConnect flag is enabled`.
+
+- A unique `tunnelIdentifier` is automatically generated for you when using the Sauce Connect Integration. An error is thrown when `tunnelIdentifier` is set manually.
+
+  Please note that if you turn `launchSauceConnect` to be true, the runner will set a unique `tunnelIdentifier` for you insead of setting by yourself. If you don't follow the rules, the runner will throw you the error: `tunnelIdentifier cannot be set when launchSauceConnect flag is enabled`.
+
+- When running Sauce Connect tunnel on your localhost, please note that Sauce Connect only supports a limited set of valid ports.
+
+  When you run your tests in your localhost, please be cautious that sauce connect only supports these [valid ports](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS#SauceConnectProxyFAQs-CanIAccessApplicationsonlocalhost?).
```
Comment (loyalchow) [inline README.md]: Please remove this, and link the first bullet line "valid ports"

**Review by loyalchow — CHANGES_REQUESTED** [review]

**Comment by xwa130saucer** [comment]
@loyalchow 
Updated all of the comments above. Sorry about letting you reword almost all of the write-ups...
I also updated the other repo (screener-storybook).

**Review by loyalchow — APPROVED** [review]

**Pull request merged**